python/cudf/cudf/core/column/column.py

```diff
@@ -1863,8 +1866,17 @@ def as_column(
         if not arbitrary.flags["C_CONTIGUOUS"]:
             arbitrary = np.ascontiguousarray(arbitrary)
 
+        delayed_cast = False
         if dtype is not None:
-            arbitrary = arbitrary.astype(np.dtype(dtype))
+            try:
+                dtype = np.dtype(dtype)
+            except TypeError:
+                # Some `dtype`'s can't be parsed by `np.dtype`
+                # for which we will have to cast after the column
+                # has been constructed.
+                delayed_cast = True
+            else:
+                arbitrary = arbitrary.astype(dtype)
```

Comment:
The increase in complexity here makes me sad, but there are so many problems with `as_column` that I don't know that it's worth our time trying to improve here.

cc: @vyasr

Reply:
I agree :cry: One day we will try to rip apart `as_column`, but today is not that day.